src/Acuminator/Acuminator.Analyzers/Analyzers/DAC/DacMethodsUsageAnalyzer.cs

```diff
@@ -0,0 +1,86 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Acuminator.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class DacMethodsUsageAnalyzer : PXDiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create
+            (
+                Descriptors.PX1031_DacCannotContainInstanceMethods,
+                Descriptors.PX1032_DacPropertyCannotContainMethodInvocations
+            );
+
+        internal override void AnalyzeCompilation(CompilationStartAnalysisContext compilationStartContext, PXContext pxContext)
+        {
+            compilationStartContext.RegisterSyntaxNodeAction(syntaxContext => AnalyzeDacDeclaration(syntaxContext, pxContext), SyntaxKind.ClassDeclaration);
+        }
+
+        private void AnalyzeDacDeclaration(SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+        {
+            ClassDeclarationSyntax classDeclaration = syntaxContext.Node as ClassDeclarationSyntax;
+            INamedTypeSymbol typeSymbol = syntaxContext.SemanticModel.GetDeclaredSymbol(classDeclaration, syntaxContext.CancellationToken);
+
+            if (typeSymbol != null && IsDac(typeSymbol, pxContext))
+            {
+                IEnumerable<MethodDeclarationSyntax> dacMethods = classDeclaration.DescendantNodes().OfType<MethodDeclarationSyntax>();
+
+                foreach (SyntaxNode node in classDeclaration.DescendantNodes())
+                {
+                    if (syntaxContext.CancellationToken.IsCancellationRequested)
+                    {
+                        return;
+                    }
+
+                    if (node is MethodDeclarationSyntax method)
+                    {
+                        AnalyzeMethodDeclarationInDac(method, syntaxContext, pxContext);
+                    }
+                    else if (node is PropertyDeclarationSyntax property)
+                    {
+                        AnalyzeMethodInvocationInDacProperty(property, syntaxContext, pxContext);
+                    }
+                }
+            }
+        }
+
+        private void AnalyzeMethodInvocationInDacProperty(PropertyDeclarationSyntax property, SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+        {
+            IEnumerable<InvocationExpressionSyntax> methodInvocations = property.DescendantNodes()
+                                                                        .OfType<InvocationExpressionSyntax>();
+            foreach (InvocationExpressionSyntax invocation in methodInvocations)
+            {
+                SymbolInfo symbol = syntaxContext.SemanticModel.GetSymbolInfo(invocation, syntaxContext.CancellationToken);
+
+                if (symbol.Symbol.Kind == SymbolKind.Method)
```

Comment:
@metallmonster symbol.Symbol could possibly be null, especialy if there was cancellation during the execution of GetSymbolInfo method. This will lead to null reference exception.